collection-scripts/gather_monitoring

```diff
@@ -5,25 +5,88 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
-BASE_COLLECTION_PATH="../must-gather"
-MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+declare -r BASE_COLLECTION_PATH="../must-gather"
+declare -r MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+declare -r CA_BUNDLE="${MONITORING_PATH}/ca-bundle.crt"
 
-mkdir -p "${MONITORING_PATH}"
+init() {
+  mkdir -p "${MONITORING_PATH}"
 
-MONITORING_ROUTE="$(oc get routes -n openshift-monitoring prometheus-k8s -o jsonpath={.status.ingress[0].host})"
-SA_TOKEN="$(oc sa get-token default)"
+  PROMETHEUS_ROUTE="$(oc get routes \
+    -n openshift-monitoring prometheus-k8s \
+    -o jsonpath='{.status.ingress[0].host}')"
 
-# this is a CA bundle we need to verify the monitoring route, we will write it to disk so we can use it in the flag
-oc -n openshift-config-managed get cm default-ingress-cert -o jsonpath='{.data.ca-bundle\.crt}' > "${MONITORING_PATH}/ca-bundle.crt"
+  ALERT_MANAGER_ROUTE="$(oc get routes \
+    -n openshift-monitoring alertmanager-main \
+    -o jsonpath='{.status.ingress[0].host}')"
 
-# using oc get --raw because we directly control it and have standardized debugging on it
-oc get\
-  --server="https://${MONITORING_ROUTE}"\
-  --token="${SA_TOKEN}" \
-  --certificate-authority="${MONITORING_PATH}/ca-bundle.crt" \
-  --raw=/api/v1/rules?type=alert 2>"${MONITORING_PATH}/alert.stderr" > "${MONITORING_PATH}/alerts.json"
+  SA_TOKEN="$(oc sa get-token default)"
 
-rm "${MONITORING_PATH}/ca-bundle.crt"
+  # this is a CA bundle we need to verify the monitoring route,
+  # we will write it to disk so we can use it in the flag
+  oc -n openshift-config-managed get cm default-ingress-cert \
+    -o jsonpath='{.data.ca-bundle\.crt}' > "$CA_BUNDLE"
+}
 
-# force disk flush to ensure that all data gathered is accessible in the copy container
-sync
+cleanup() {
```

Comment:
Is this needed? When the 'container' is destroyed (at the completion of a gather) won't this also be deleted then? Is this a wasted cycle?

Reply:
I wasn't sure either if the cleanup is really needed but I left it there because 
1. I would consider as best practice for each script to cleanup what it created so that it doesn't influence/affect other scripts that gets executed after this one
2. This was already present in the original implementation

Reply:
On  a closer look, yes, we want to delete the CA bundle since that isn't something we want to gather.